contracts/pcv/utils/NamedStaticPCVDepositWrapper.sol

```diff
@@ -0,0 +1,201 @@
+    event DepositRemoved(uint256 index);
+    
+    /// @notice event to add a new deposit
+    event DepositAdded(uint256 index, string indexed depositName);
+
+    /// @notice event emitted when a deposit is edited
+    event DepositChanged(uint256 index, string indexed depositName);
+
+    /// @notice struct to store info on each PCV Deposit
+    struct DepositInfo {
+        string depositName;
+        uint256 usdAmount; /// USD equivalent in this deposit, not including FEI value
+        uint256 feiAmount; /// amount of FEI in this deposit
+        uint256 underlyingTokenAmount; /// amount of underlying token in this deposit
+        address underlyingToken; /// address of the underlying token this deposit is reporting
+    }
+
+    /// @notice a list of all pcv deposits
+    DepositInfo[] public pcvDeposits;
+
+    /// @notice the PCV balance
+    uint256 public override balance;
+
+    /// @notice the reported FEI balance to track protocol controlled FEI in these deposits
+    uint256 public feiReportBalance;
+
+    constructor(address _core, DepositInfo[] memory newPCVDeposits) CoreRef(_core) {
+
+        // Uses oracle admin to share admin with CR oracle where this contract is used
+        _setContractAdminRole(keccak256("ORACLE_ADMIN_ROLE"));
+
+        // add all pcv deposits
+        for (uint256 i = 0; i < newPCVDeposits.length; i++) {
+            _addDeposit(newPCVDeposits[i]);
+        }
+    }
+
+    // ----------- Helper methods to change state -----------
+
+    /// @notice helper method to add a PCV deposit
+    function _addDeposit(DepositInfo memory newPCVDeposit) internal {
+        require(newPCVDeposit.feiAmount > 0 || newPCVDeposit.usdAmount > 0, "NamedStaticPCVDepositWrapper: must supply either fei or usd amount");
+
+        uint256 oldBalance = balance;
+        uint256 oldFEIBalance = feiReportBalance;
+
+        balance += newPCVDeposit.usdAmount;
+        feiReportBalance += newPCVDeposit.feiAmount;
+        pcvDeposits.push(newPCVDeposit);
+
+        emit DepositAdded(pcvDeposits.length - 1, newPCVDeposit.depositName);
+        emit BalanceUpdate(oldBalance, balance, oldFEIBalance, feiReportBalance);
+    }
+
+    /// @notice helper method to edit a PCV deposit
+    function _editDeposit(
+        uint256 index,
+        string calldata depositName,
+        uint256 usdAmount,
+        uint256 feiAmount,
+        uint256 underlyingTokenAmount,
+        address underlyingToken
+    ) internal {
+        require(index < pcvDeposits.length, "NamedStaticPCVDepositWrapper: cannot edit index out of bounds");
+
+        DepositInfo storage updatePCVDeposit = pcvDeposits[index];
+
+        uint256 oldBalance = balance;
+        uint256 oldFEIBalance = feiReportBalance;
+
+        balance = balance - updatePCVDeposit.usdAmount + usdAmount;
+        feiReportBalance = feiReportBalance - updatePCVDeposit.feiAmount + feiAmount;
+
+        updatePCVDeposit.usdAmount = usdAmount + updatePCVDeposit.usdAmount;
```

Comment:
is the feiAmount getting updated? If no we need to fix and add a unit test

Reply:
Good catch!